src/GafferOSL/OSLImage.cpp

```diff
@@ -148,6 +182,20 @@ void OSLImage::affects( const Gaffer::Plug *input, AffectedPlugsContainer &outpu
 		outputs.push_back( outPlug()->channelNamesPlug() );
 		outputs.push_back( outPlug()->channelDataPlug()	);
 	}
+	else if( input == inPlug()->formatPlug() )
+	{
+		outputs.push_back( outPlug()->formatPlug() );
+		outputs.push_back( shadingPlug() );
+	}
+	else if( input == inPlug()->dataWindowPlug() )
+	{
+		outputs.push_back( outPlug()->dataWindowPlug() );
+	}
+	else if( defaultFormatPlug()->isAncestorOf( input ) )
```

Comment:
Avoid declaring dependencies indirectly like this - we don't pull on `defaultFormatPlug()` when computing `out.format` or `out.dataWindow`. Declare the true dependency instead - that is, the one to our `__defaultInput` plug (the one I proposed in another comment).